components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-list.jsp

```diff
@@ -122,7 +123,11 @@
                         <tr>
                             <td colspan="3"><i><fmt:message key='no.idp'/></i></td>
                         </tr>
-                    <% } %>
+                    <% }
+                        if (session.getAttribute(identityProvider) != null) {
+                            session.removeAttribute(identityProvider);
+                        }
```

Comment:
Shall we move this to the top, may be before line 33 ? And also no need to do null check here right.. ? doesn't harm even if the value is null.